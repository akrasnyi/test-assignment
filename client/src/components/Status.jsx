import React from 'react'
import styled from 'styled-components'
import PropTypes from 'prop-types'

const StyledContainer = styled.div`
  display: flex
  flex-direction: column
  align-items: center
  margin-top: 50px;
  border: 1px solid #827b7b;
  border-radius: 5px;
  padding: 20px;
`

function Status(props) {
  const { fname, lname, email, iban } = props.fields
  return (
    <React.Fragment>
      <StyledContainer>
        <h2>Input Status</h2>
        <p>
          Current First Name: <b>{fname}</b>
        </p>
        <p>
          Current Last Name: <b>{lname}</b>
        </p>
        <p>
          Current Email: <b>{email}</b>
        </p>
        <p>
          Current IBAN: <b>{iban}</b>
        </p>
      </StyledContainer>
    </React.Fragment>
  )
}

Status.propTypes = {
  fields: PropTypes.shape({
    fname: PropTypes.string,
    lname: PropTypes.string,
    email: PropTypes.string,
    iban: PropTypes.string
  })
}

export default Status
