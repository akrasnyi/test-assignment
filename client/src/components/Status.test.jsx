import React from 'react'
import Status from './Status'
import renderer from 'react-test-renderer'

const fields = {
  fname: '',
  lname: '',
  email: '',
  iban: 'UA903052992990004149123456789' // test value, valid iban, can be ommited
}

it('renders correctly', () => {
  const tree = renderer.create(<Status fields={fields} />).toJSON()
  expect(tree).toMatchSnapshot()
})
