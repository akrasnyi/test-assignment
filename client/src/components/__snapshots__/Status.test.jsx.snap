// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  className="sc-bdVaJa PybKf"
>
  <p>
    Current First Name: 
    <b>
      
    </b>
  </p>
  <p>
    Current Last Name: 
    <b>
      
    </b>
  </p>
  <p>
    Current Email: 
    <b>
      
    </b>
  </p>
  <p>
    Current IBAN: 
    <b>
      UA903052992990004149123456789
    </b>
  </p>
</div>
`;
